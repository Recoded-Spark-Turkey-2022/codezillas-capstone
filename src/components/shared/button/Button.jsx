/* eslint-disable react/destructuring-assignment */
import { useNavigate } from "react-router-dom";

export default function Button(props)
{
    const navigate = useNavigate();

    function handleClick()
    {
        if(props.function)
        {
            props.function()
        }else{
            navigate('/')
        }
    }
    return(
        <input 
        type={props.type}
        value={props.name}
        className={props.classes}
        onClick={()=> handleClick()} />
    );
}