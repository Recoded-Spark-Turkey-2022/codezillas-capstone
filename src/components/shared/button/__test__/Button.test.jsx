import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Button from '../Button';
import { store } from '../../../../app/store';

it('Navbar component renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Button />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
