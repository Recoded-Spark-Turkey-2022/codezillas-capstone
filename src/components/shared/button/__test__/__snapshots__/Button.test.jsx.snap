// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly 1`] = `
<input
  className="py-2 px-5 w-10/12 md:w-8/12 lg:w-5/12 hover:bg-cyan-200 bg-cyan-400 font-medium rounded text-sm md:text-lg"
  onClick={[Function]}
  type="button"
  value="BACK TO HOME"
/>
`;
